### PR TITLE
US-ShareNews-02: Rework news share design

### PR DESCRIPTION
**Pull request by aelbenney**
This change replace news share popup with a drawer and also synchronize data in other news views to well show the news preview in the share drawer. Besides this will fix illustration in shared news.

### CODE REVIEW

File: webapp/src/main/webapp/news-details/components/ExoNewsShareActivity.vue [hunk 1]
```diff
@@ -38,28 +80,16 @@ import * as newsServices from '../../services/newsServices';
...
       required: true,
-      default: false
+      default: function() { return new Object(); }
     }
   },
   data() {
     const self = this;
     return {
-      showShareNewsPopup: false,
+      open: false,
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/news-details/components/ExoNewsShareActivity.vue]: We should keep a more explicit name IMO, for example showShareNewsDrawer. WDYT ?

File: webapp/src/main/webapp/news/components/NewsApp.vue [hunk 2]
```diff
@@ -54,19 +54,21 @@
         color="#578dc9" /> </v-app>
     <div v-if="newsList.length" id="newsListItems" class="newsListItems">
       <div v-for="news in newsList" :key="news.newsId" class="newsItem">
-        <a :href="news.newsUrl" :style="{ 'background-image': 'url(' + news.newsIllustration + ')' }" class="newsItemIllustration"></a>
+        <a :href="news.url" :style="{ 'background-image': 'url(' + news.illustrationURL + ')' }" class="newsItemIllustration"></a>
+        <!--
+        <img :src="news.illustrationURL" class="newsItemIllustration" alt="News"/>-->
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/news/components/NewsApp.vue]: To remove.

File: webapp/src/main/webapp/groovy/news/webui/activity/UISharedNewsFullActivity.gtmpl [hunk 3]
```diff
@@ -163,6 +163,8 @@ if (activity && news) {
   String illustrationURL = null;
   if(news.illustration == null) {
     illustrationURL = new StringBuilder("/news/images/newsImageDefault.png").toString();
+  } else {
+     illustrationURL = news.illustrationURL;
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/groovy/news/webui/activity/UISharedNewsFullActivity.gtmpl]: The default illustration must not be displayed if there is no illustration, should it be:
```
  String illustrationURL = null;
  if(news.illustrationURL != null) {
    illustrationURL = new StringBuilder("/rest/v1/news/").append(news.id).append("/illustration").toString();
  }
```
?

File: webapp/src/main/webapp/news-details/components/ExoNewsShareActivity.vue (hunk 1 above)
Comment (aelbenney) [reply webapp/src/main/webapp/news-details/components/ExoNewsShareActivity.vue]: the open is used for the drawer className to whether put "open" word or not.

Comment (thomasdelhomenie) [reply webapp/src/main/webapp/news-details/components/ExoNewsShareActivity.vue]: Yes, it allows to have a shorter syntax for the class, but it makes the variable less explicit IMO. I would prefer something like https://github.com/exo-addons/news/blob/develop/webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue#L87

**Comment by thomasdelhomenie** [comment]
Replaced by https://github.com/exo-addons/news/pull/163

**Pull request closed without merging**